---
title: "HW5"
author: "Brandon Turner"
date: "4/12/2021"
output: rmarkdown::github_document
---

## 1)	We will utilize our RNA seq dataset of E. Coli genes from mice.  Read and normalize the counts table ( “nc101_scaff_dataCounts.txt “ into R).

```{r}
# read in data
myData = read.table('/Users/bturne48/Documents/GitHub/AdvStats/HW5/longitdunalRNASeqData/nc101_scaff_dataCounts.txt', header = T, row.names = 1, sep ='\t', colClasses = c('character', rep('numeric', 11)))

# remove rare genes
myData = myData[apply(myData, 1, median) > 5, ]

# normalize
myData_norm = myData
for (i in 1:ncol(myData)){
  colSum = sum(myData[,i])
  myData_norm[,i] = myData_norm[,i]/colSum
}
head(myData_norm)

```

## 2)	For every row in the normalized spreadsheet, run three t-tests ( “day 2” vs. “week 12”, “day 2” vs. “week 18” and “week 12” vs. “week 18”.  At a p < .05 threshold fill in the following table:

```{r}
# uncorrected ounts
y1_uncor = 0
y1_pval = c()
y2_uncor = 0
y2_pval = c()
y3_uncor = 0
y3_pval = c()
for (i in 1:nrow(myData_norm)){
  # day 2 vs week 12
  y1_pval[i] = t.test(myData_norm[i,1:3], myData_norm[i,4:6])$p.value
  if (y1_pval[i] <= 0.05) {
    y1_uncor = y1_uncor + 1
  }
  
  # day 2 vs week 18
  y2_pval[i] = t.test(myData_norm[i,1:3], myData_norm[i,7:11])$p.value
  if (y2_pval[i] <= 0.05) {
    y2_uncor = y2_uncor + 1
  }
  
  # week 12 vs week 18
  y3_pval[i] = t.test(myData_norm[i,4:6], myData_norm[i,7:11])$p.value
  if (y3_pval[i] <= 0.05) {
    y3_uncor = y3_uncor + 1
  }
  
}
uncorrected_res = c(y1_uncor, y2_uncor, y3_uncor)


# FDR corrected
x = sum(p.adjust(y1_pval, 'BH') < 0.05)
y = sum(p.adjust(y2_pval, 'BH') < 0.05) 
z = sum(p.adjust(y3_pval, 'BH') < 0.05)
fdr_corrected_res = c(x,y,z)


# Bon corrected
bonf_corrected_res = c(sum(p.adjust(y1_pval, "bonferroni") <= 0.05), sum(p.adjust(y2_pval, "bonferroni") <= 0.05), sum(p.adjust(y3_pval, "bonferroni") <= 0.05))


# results df 
endTable = data.frame(uncorrected_res, fdr_corrected_res, bonf_corrected_res)
endTable
```

## 3)	Make histograms of all the uncorrected p-values for each of the three companions.  Are any of the distributions uniform?

```{r}
# use breaks=20 so that the counts field in the histogram can be parsed (hist[1:2] = all pvals < 0.05)
hist1 = hist(y1_pval, main ='Day 2 vs Week 12', breaks = 20)
hist2 = hist(y2_pval, main ='Day 2 vs Week 18', breaks = 20)
hist3 = hist(y3_pval, main ='Week 12 vs Week 18', breaks = 20)
```

None of the histrograms appear to show a unif dist

## 4) Based on these data, when is the biggest shift in the transcriptome?  Which samples are most different from one another?
```{r}
# sum all pvals below 0.05 for all 3 comparisons
sum(hist1$counts[1:2])
sum(hist2$counts[1:2])
sum(hist3$counts[1:2])
```


Day 2 vs Week 18 shows the most values that show p < 0.05. 